From a640a9ace70fc6cbff965bb742917bf8b9d3a8ea Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@nxp.com>
Date: Thu, 28 Sep 2017 10:58:24 +0000
Subject: [PATCH 517/706] staging: fsl-dpaa2/mac: Remove redundant free

free_netdev (put_device) already handles freeing the private data
structure, and KASAN will complain due to a free after free if we
explicitly do the same afterwards.

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@nxp.com>
(cherry picked from commit aa641f018706e9a926dbda1b3d555e430c7237d3)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 96ca54bce94b..ecfc8a9841ce 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -639,7 +639,6 @@ static int dpaa2_mac_remove(struct fsl_mc_device *mc_dev)
 	free_netdev(priv->netdev);
 
 	dev_set_drvdata(dev, NULL);
-	kfree(priv);
 
 	return 0;
 }
-- 
2.17.1

